spec/integration/action/format_config_spec.rb

```diff
@@ -137,12 +137,26 @@ def handle(req, res)
     write "config/app.rb", <<~RUBY
       require "hanami"
 
+      class CustomBodyParser
+        def mime_types
+          ['application/json']
+        end
+
+        def parse(body, *)
+          JSON.parse(body).transform_values { |v| v.map(&:upcase) }
+        end
+
+        def empty?
```

Comment:
I'm not sure why it keeps calling `empty?` on a custom parser. Without this method I get this:

     NoMethodError:
       undefined method `empty?' for #<CustomBodyParser:0x000000010bb4f170>

                 return DEFAULT_BODY_PARSERS if parser_specs.empty?
                                                            ^^^^^^^
     # ./lib/hanami/slice/routing/middleware/stack.rb:128:in `new'
     # ./lib/hanami/slice/routing/middleware/stack.rb:128:in `to_rack_app'

I'll try to investigate what's going on but if you have any idea please let me know!

Reply:
figured that out